src/main/java/com/sequenceiq/cloudbreak/service/stack/connector/aws/AwsConnector.java

```diff
@@ -177,6 +163,62 @@ public void buildStack(Stack stack, String userData, Map<String, Object> setupPr
         }
     }
 
+    private String getEbsSnapshotIdIfNeeded(Stack stack, Long stackId) {
+        if (isEncryptedVolumeRequested(stack)) {
+            Optional<String> snapshot = createSnapshotIfNeeded(stack);
+            if (snapshot.isPresent()) {
+                return snapshot.orNull();
+            } else {
+                throw new CloudFormationStackException(String.format("Failed to create Ebs encrypted volume on stack: %s", stackId));
+            }
+        } else {
+            return null;
+        }
+    }
+
+    private Optional<String> createSnapshotIfNeeded(Stack stack) {
+        AmazonEC2Client client = awsStackUtil.createEC2Client(stack);
+        DescribeSnapshotsRequest describeSnapshotsRequest = new DescribeSnapshotsRequest()
+                .withFilters(
+                        new Filter().withName("tag-key").withValues(CLOUDBREAK_EBS_SNAPSHOT),
+                        new Filter().withName("tag-value").withValues(CLOUDBREAK_EBS_SNAPSHOT));
```

Comment:
it doesn't really matter but these 2 filters would return all of the following tags:
```
CLOUDBREAK_EBS_SNAPSHOT: CLOUDBREAK_EBS_SNAPSHOT
CLOUDBREAK_EBS_SNAPSHOT: some other value
some other key: CLOUDBREAK_EBS_SNAPSHOT
```
One of them is enough or the `"tag :key =*value*"` filter can be used